source/source_integration_test.go

```diff
@@ -211,6 +216,95 @@ func TestSource_Snapshot_Continue(t *testing.T) {
 	is.NoErr(err)
 }
 
+func TestSource_CDC_Success(t *testing.T) {
+	t.Parallel()
+
+	is := is.New(t)
+
+	ctx := context.Background()
+
+	tableName := "conduit_source_cdc_integration_test_success"
+
+	cfg := prepareConfig(t, tableName)
+	cfg[ConfigKeyColumns] = "int_column,text_column"
+
+	err := prepareData(
+		ctx, cfg[config.KeyAddress], cfg[config.KeyKeyspace], cfg[config.KeyTabletType], tableName, true,
+	)
+	is.NoErr(err)
+
+	t.Cleanup(func() {
+		err = clearData(ctx, cfg[config.KeyAddress], cfg[config.KeyKeyspace], cfg[config.KeyTabletType], tableName)
+		is.NoErr(err)
+	})
+
+	s := new(Source)
+
+	err = s.Configure(ctx, cfg)
+	is.NoErr(err)
+
+	// wait a bit for Vitess to get the "create table" query results from each shard
+	time.Sleep(time.Second * 3)
```

Comment:
is it 3 second enough?
What will be if it takes tome more than 3 second?

Is it possible to do it without timeout?

Reply:
If it takes more than 3 seconds we will get an error saying that a table doesn't exist. But in my experience, it never took longer than 3 seconds.

Reply:
maybe we can check if table exist in loop with timeout. What do you think about it?

Reply:
I've rewrote the `prepareData` function, now it creates test tables using `vtgateconn` and `session` and it seems that these things wait for a table to be created internally